### PR TITLE
Added validation and fixed handling of trailing slashes in URLs

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #7767

### CODE REVIEW

File: pulp_ansible/app/serializers.py [hunk 1]
```diff
@@ -128,10 +129,23 @@ def validate(self, data):
             rest_framework.serializers.ValidationError: If the url or requirements_file is invalid
 
         """
+
+        def _validate_url(url):
+            """Based on https://git.io/JTMAA."""
+            if not url.endswith("/") and url != "https://galaxy.ansible.com":
```
Comment (fao89) [inline pulp_ansible/app/serializers.py:135]: maybe we need a constant with URLs, I believe we need to add https://galaxy-dev.ansible.com

File: pulp_ansible/app/tasks/collections.py [hunk 2]
```diff
@@ -336,18 +336,29 @@ async def _fetch_collections(self):
...
-                root = f"{root}/api/"
+                if root.endswith("/api/"):
+                    raise
+
+                root = urljoin(root, "api/")
                 downloader = remote.get_downloader(url=root)
                 api_data = parse_metadata(await downloader.run())
 
+            if "available_versions" not in api_data:
+                raise RuntimeError(_("Could not find 'available_versions' at {}").format(root))
```
Comment (fao89) [inline pulp_ansible/app/tasks/collections.py:360]: this error message is pretty helpful, I like it!

File: pulp_ansible/tests/functional/api/collection/v2/test_sync.py [hunk 3]
```diff
@@ -45,6 +44,33 @@ def setUpClass(cls):
...
         cls.client = api.Client(cls.cfg)
 
+    def test_remote_urls(self):
+        """This tests that the remote url ends with a "/".
+
+        This test targets https://pulp.plan.io/issues/7767
+
+        ansible-galaxy requires a trailing slash: https://git.io/JTMAA
+
+        It does make an exception for "https://galaxy.ansible.com": https://git.io/JTMpk
```
Comment (fao89) [inline pulp_ansible/tests/functional/api/collection/v2/test_sync.py:54]: awesome!

**Review by fao89 — COMMENTED** [review]
LGTM! Just waiting for my vagrant provisioning so I can test AH

File: pulp_ansible/app/serializers.py (hunk 1 above)
Comment (daviddavis) [reply pulp_ansible/app/serializers.py:135]: In order to do that, I'd have to make [this logic](https://github.com/pulp/pulp_ansible/pull/403/files#diff-e6f9543dbfb3c4a3c4c71d8068b87a2388ef384e16d8b19fc61ec7c5e03d7919R344-R345) more complex. I don't think many users will use dev galaxy and if they do, they can use `https://galaxy-dev.ansible.com/api/` which is a link to the actual api.

Comment (daviddavis) [reply pulp_ansible/app/serializers.py:135]: This line is just sort of a nasty hack that I have to carry over from ansible-galaxy. See https://github.com/ansible/ansible/blob/a51a6f4a259b45593c3f803737c6d5d847258a83/lib/ansible/galaxy/api.py#L52-L53.

**Review by fao89 — APPROVED** [review]
Just tested and it fixes the CI

**Pull request merged**